GirlsFlix/src/main/java/com/gfx/service/Scheduler.java

```diff
@@ -0,0 +1,56 @@
+package com.gfx.service;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import com.gfx.Config;
+
+public class Scheduler extends Thread{
+	private static int count = 0;
+	private SerieService serieService;
+	private SerieFactory serieFactory;
+	private boolean firstTimerIteration = true;
+	
+	public Scheduler() {
+		this.start();
+	}
+	
+	public void run() {
+		serieService = new SerieService();
+		serieFactory = new SerieFactory();
+		Timer timer = new Timer();
+		Date initDate = convertLocalDateTimeToDate(Config.initDateScheduler);
+		//launch the timer at the date initDate and excute the method run of the ReccurentTask class every Config.internalForScheduler milliseconds
+		timer.scheduleAtFixedRate(new ReccurentTask(), initDate, Config.intervalForScheduler);
+	}
+	
+	public Date convertLocalDateTimeToDate(LocalDateTime localDateTime) {
+		return Date.from(localDateTime.atZone(ZoneId.systemDefault()).toInstant());
+	}
+	
+	public class ReccurentTask extends TimerTask{
+		
+		
+		public void run() {
+			count++;
+			System.out.println("itération du scheduler : " + count);
+			if(firstTimerIteration) {
```

Comment:
ça permet de ne pas lancer la MàJ de la BDD lors du premier lancement du server c'est ça ?

Reply:
oui c'est ça à la première connexion on ne fait pas de connexion à l'API. Ensuite à partir de la deuxième itération on fait l'appel à l'API et on met à jour la BDD et les objets